### ISSUE TITLE
CenterOverAssociatedObject="True" may "hang" the software

### ISSUE THREAD

**Issue by znakeeye** [PrismLibrary/Prism#174]
Using an InteractionRequestTrigger and a PopupWindowAction, modal dialogs may end up outside the desktop. Consider the popup window action below. If the triggers are associated with a control near the bottom/right of the screen, the popup window may be placed completely outside the desktop. This hangs the software if the user does not know about the "Press Alt+Space to move hidden modal dialog" trick...

Should be fairly easy to fix. In *PopupWindowAction.cs* we should apply some clamping to the wrapper window's coordinates:

```c#
wrapperWindow.Top = position.Y + ((view.ActualHeight - wrapperWindow.ActualHeight) / 2); 
wrapperWindow.Left = position.X + ((view.ActualWidth - wrapperWindow.ActualWidth) / 2); 

// TODO: Clamp to screen!
```

**CenterOverAssociatedObject** is not 100% usable :(

```xaml
<i:Interaction.Triggers>
	<prism:InteractionRequestTrigger SourceObject="{Binding ConfirmationRequest, Mode=OneWay}">
		<prism:PopupWindowAction IsModal="True" CenterOverAssociatedObject="True">
			<prism:PopupWindowAction.WindowContent>
				<local:ConfirmationView />
			</prism:PopupWindowAction.WindowContent>
		</prism:PopupWindowAction>
	</prism:InteractionRequestTrigger>
</i:Interaction.Triggers>
```

**Comment by brianlagunas** [comment]
Do you have a sample that can demonstrate this issue that we could look at?

**Comment by brianlagunas** [comment]
Is this related to #175 and will the proposed fix solve your problem?

**Comment by brianlagunas** [comment]
Can you provide us with anymore information around your scenario to help replicate the issue?  Are you wanting the dialog to be centered in the parent window, or an element within the view that is located on the bottom right?

**Comment by znakeeye** [comment]
I'm trying to center the dialog in the parent element, which sometimes is an element located on the bottom right. More specifically:

I have an ItemControl with an item template consisting of a button that shows a modal dialog. It makes sense to show the dialog near this button, which is why I tried to center it using the CenterOverAssociatedObject property. Then there is this problem when clicking the button in the bottom of the list...

**Comment by brianlagunas** [comment]
Can you provide a sample that shows this behavior?  I haven't been able to get it to go off screen.

**Comment by znakeeye** [comment]
How soon do you need it in order to get a possible fix in the next release?

**Comment by brianlagunas** [comment]
fixed with #245
